dask/dataframe/multi.py

```diff
@@ -867,6 +876,10 @@ def merge_asof(
     result = merge_asof_indexed(left, right, **kwargs)
     if left_on or right_on:
         result = result.reset_index()
+        if ixcol is not None:
+            result = result.set_index(ixcol)
+            result._meta = result._meta.rename_axis(ixname)
```

Comment:
This line should be unnecessary, the `map_partitions` call should handle this for you.

Reply:
seems it does